src/layout/componentValidation.ts

```diff
@@ -0,0 +1,79 @@
+import { implementsValidateComponent, implementsValidateEmptyField, implementsValidateSchema } from '.';
+
+import { FrontendValidationSource } from 'src/features/validation';
+import { runExpressionValidationsOnNode } from 'src/features/validation/frontend/expressionValidation';
+import { isComponentValidation, isFieldValidation } from 'src/features/validation/utils';
+import type {
+  ComponentValidation,
+  FieldValidation,
+  FormValidations,
+  ISchemaValidationError,
+  ValidationDataSources,
+} from 'src/features/validation';
+import type { CompTypes } from 'src/layout/layout';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+// TODO(Validation): Merge all frontend validations into one group?
```

Comment:
Still relevant?

Reply:
Yes, after cleaning up the frontend validation implementation I realized that all frontend validations are always run together now anyway, so there is no need to separate the frontend sources (which would be necessary if we for example sometimes skipped required validation). But as the first note says, the group structure will likely need to be changed to accommodate the coming changes to backend validations anyways. So I will rather deal with it then.